integration-test/erc1155mintable.contract.test-integration.js

```diff
@@ -0,0 +1,175 @@
+      baseURI: 'https://www.cryptotimes.io/wp-content/uploads/2022/03/BAYC-835-Website-800x500.jpg',
+    });
+    const receipt = await tx.wait();
+    expect(receipt.status).toEqual(1);
+  });
+
+  it('should mint nft', async () => {
+    const tx = await contractObject.mint({
+      to: owner,
+      id: 0,
+      quantity: 1,
+    });
+
+    const receipt = await tx.wait();
+    expect(receipt.status).toEqual(1);
+  });
+
+  it('should set approval for all', async () => {
+    const loadedContractObject = await sdk.loadContract({
+      template: TEMPLATES.ERC1155Mintable,
+      contractAddress: contractObject.contractAddress,
+    });
+    const tx = await loadedContractObject.setApprovalForAll({
+      to: publicAddress,
+      approvalStatus: true,
+    });
+    const receipt = await tx.wait();
+
+    expect(receipt.status).toEqual(1);
+  });
+
+  it('should transfer nft with approval', async () => {
+    // owner mints a token to themselves
+    const tx = await contractObject.mint({
+      to: owner,
+      id: 0,
+      quantity: 1,
+    });
+
+    await tx.wait();
+
+    // owner approves publicAddress to transfer token that he owns
+    const txApprove = await contractObject.setApprovalForAll({
+      to: publicAddress,
+      approvalStatus: true,
+    });
+
+    await txApprove.wait();
+
+    const accountPublic = new Auth({
+      privateKey: privateKeyPublicAddress,
+      projectId: process.env.INFURA_PROJECT_ID,
+      secretId: process.env.INFURA_PROJECT_SECRET,
+      rpcUrl: 'http://0.0.0.0:8545',
+      chainId: 5,
+    });
+
+    const sdkPublic = new SDK(accountPublic);
+
+    const existing = await sdkPublic.loadContract({
+      template: TEMPLATES.ERC1155Mintable,
+      contractAddress: contractObject.contractAddress,
+    });
+
+    // publicAddress transfers token of owner
+    const txTransfer = await existing.safeTransferFrom({
+      from: owner,
+      to: thirdUser,
+      id: 0,
+      amount: 1,
+    });
+
+    const receipt = await txTransfer.wait();
+
+    expect(receipt.status).toEqual(1);
+  });
+
+  it('should return setRoyalties', async () => {
+    await contractObject.royalties.setRoyalties({ publicAddress, fee: 1000 });
+    const infos = await contractObject.royalties.royaltyInfo({ tokenId: 1, sellPrice: 10 });
+
+    expect(infos).toStrictEqual([utils.getAddress(publicAddress), BigNumber.from('1')]);
+  });
+
+  it('should return setRoyalties when tokenId is zero', async () => {
+    await contractObject.royalties.setRoyalties({ publicAddress, fee: 1000 });
+    const infos = await contractObject.royalties.royaltyInfo({ tokenId: 0, sellPrice: 10 });
+
+    expect(infos).toStrictEqual([utils.getAddress(publicAddress), BigNumber.from('1')]);
+  });
+
+  it('should renounce contract ownership', async () => {
+    const result = await contractObject.accessControl.renounceOwnership();
+    const receipt = await result.wait();
+
+    expect(receipt.status).toBe(1);
+  });
+});
```

Comment:
i think its better to follow the same logic and split the integration tests , we can create folders for each template